Sources/SwiftInspectorVisitors/NestableTypeVisitor.swift

```diff
@@ -88,9 +88,18 @@ public final class NestableTypeVisitor: SyntaxVisitor {
   }
 
   public override func visit(_ node: TypealiasDeclSyntax) -> SyntaxVisitorContinueKind {
-    let typealiasVisitor = TypealiasVisitor(parentType: parentType)
-    typealiasVisitor.walk(node)
+    guard
+      !topLevelParsingTracker.hasFinishedParsing,
+      let topLevelDeclarationName = topLevelDeclaration?.nestableInfo.name
+    else {
+      assertionFailure("Encountered more than one top-level declaration. This is a usage error: a single NestableTypeVisitor instance should start walking only over a declaration syntax node")
+      return .skipChildren
+    }
 
+    // We've previously found an top-level declaration, so this must be an inner declaration.
+    let newParentType = TypeDescription(name: topLevelDeclarationName, parent: parentType)
```

Comment:
To make sure I follow, this _is_ the type we are visiting right?

Reply:
This is the name of the nestable type we're in the process of visiting, which is the parent type of this typealias we just found